build/packaging/pre_build_script.sh

```diff
@@ -16,6 +16,7 @@ set -euxo pipefail
 readonly BUILD_DEPS=(
   # This list must match the build-system.requires list from pyproject.toml.
   "cmake"
+  "pip>=23"
```

Comment:
In case you couldn't set this for older python versions, I wonder if this syntax works:

```
pip ; python_version < "3.10"
pip>=23 ; python_version >= "3.10"
```

https://github.com/pytorch/pytorch/blob/main/.ci/docker/requirements-ci.txt#L220-L221

Building ET with older python version is expected to fail atm IIRC